### ISSUE TITLE
Link component does not automatically keep the same language when switching pages in app directory

### ISSUE THREAD

**Issue by leosollereder** [vercel/next.js#54880]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:22 PDT 2023; root:xnu-8796.121.3~7/RELEASE_X86_64
    Binaries:
      Node: 16.14.0
      npm: 8.3.1
      Yarn: 1.22.19
      pnpm: N/A
    Relevant Packages:
      next: 13.4.20-canary.15
      eslint-config-next: 13.4.19
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.2.2
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Internationalization (i18n), Middleware / Edge (API routes, runtime)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/leosollereder/nextI18n

### To Reproduce

To reproduce the bug, you need to follow the official documentation on i18n in the app router:

https://nextjs.org/docs/app/building-your-application/routing/internationalization

### Describe the Bug

The code uses two different languages, "en" and "de", with "de" being the defaultLanguage. 
When switching the language to english, and then using a <Link> component to another page, in the above case "/testsite", it leads me to the "/de/testsite" page, instead of keeping the english version.
Is this the expected/desired behaviour, and do I manually need to add the language to each href in the <Link> components for this to work?

### Expected Behavior

The expected behaviour would be, that upon enabling localization, the language stays the same, even when switching pages, without having to manually add it at the beginning of a href of a <Link>.

**Comment by eben-vranken** [comment]
I suppose since German is your default it will route to those pages instead of the ones currently active. 

A solution I would recommend is keeping user language preferences in local storage (or even better, as actual user data) and using this as an "in-between" when routing, that way you wouldn't ever need to set href manually.

Example:

```
// Check for language preferences in localStorage
let languagePreference = localStorage.getItem("languagePreference ")

if(!languagePref) {
   languagePreference = "de";
   localStorage.setItem("languagePref", languagePreference );
}

<Link href={`${languagePreference}/your-route`}>Example</Link>
```

Then of course you would have to update this when people change language preferences, but should be quite straightforward

**Comment by leosollereder** [comment]
1.

**Comment by leosollereder** [comment]
> I suppose since German is your default it will route to those pages instead of the ones currently active.
> 
> A solution I would recommend is keeping user language preferences in local storage (or even better, as actual user data) and using this as an "in-between" when routing, that way you wouldn't ever need to set href manually.
> 
> Example:
> 
> ```
> // Check for language preferences in localStorage
> let languagePreference = localStorage.getItem("languagePreference")
> 
> if(!languagePreference) {
>    languagePreference = "de";
>    localStorage.setItem("languagePreference", languagePreference);
> }
> 
> <Link href={`${languagePreference}/your-route`}>Example</Link>
> ```
> 
> Then of course you would have to update this when people change language preferences, but should be quite straightforward


While this does solve the problem, I wonder, if that is the official solution which is recommended by nextJS, or if there is a different way one should approach this.

**Comment by eben-vranken** [comment]
I'm also very curious. Although, like you said, the solution I provided could work (and would be a simple fix) but it would be nice if there were (could be that it already exists) a native Next solution since I can imagine multi-language routing is very common

**Comment by mark-wff** [comment]
@eben-vranken, yes, in the previous Next.js versions, it was simple to implement multi-language routing. However, in Next.js 13, there's no obvious way to make things work. It's possible that there's just a missing part in the documentation.

@timneutkens: It would be great to have some support from the Next.js team here.

**Comment by eben-vranken** [comment]
Please let me know in case you find something
